rllib/core/rl_trainer/trainer_runner.py

```diff
@@ -123,27 +165,40 @@ def _distributed_update(self, batch: MultiAgentBatch) -> List[Mapping[str, Any]]
             different backend than the cuda backend.
 
         Args:
-            batch: The data to use for the update.
+            See `.update()` docstring.
 
         Returns:
             A list of dictionaries of results from the updates from the RLTrainer(s)
         """
         refs = []
         global_size = len(self._workers)
-        batch_size = math.ceil(len(batch) / global_size)
         for i, worker in enumerate(self._workers):
             batch_to_send = {}
             for pid, sub_batch in batch.policy_batches.items():
                 batch_size = math.ceil(len(sub_batch) / global_size)
                 start = batch_size * i
                 end = min(start + batch_size, len(sub_batch))
                 batch_to_send[pid] = sub_batch[int(start) : int(end)]
+            # TODO (Avnish): int(batch_size) ? How should we shard MA batches really?
```

Comment:
maybe we can start with however we were doing things before in the multi gpu training code

Reply:
each policy was responsible for sharding its own sample then. But now we have control over the full MARLModule's training in each RLTrainer actor and sharding is happening before that. So the old way is not transferable.